### PR TITLE
Patch import projet from Gitlab

### PR DESCRIPTION
**Pull request by Mickael-Martin**
Please check if you are no regression. It works for me with url of project : https://framagit.org/Mickael-Martin/zabbix_ynh or url of branch : https://framagit.org/Mickael-Martin/zabbix_ynh/tree/testing

## The problem

Cannot import Gitlab (an other gitlab based solutions) project branch.
...

## Solution

Use correct git syntax with correct url
...

## PR Status

...

## How to test

...

## Validation

- [ ] Principle agreement 0/2 : 
- [ ] Quick review 0/1 : 
- [ ] Simple test 0/1 : 
- [ ] Deep review 0/1 :

### CODE REVIEW

**Comment by Psycojoker** [comment]
Hello,

I'm wondering if we should do something like :

```python
if 'github.com/' in url: 
    # current behavior
else:
    # your patch
```

That would be a safer approach.

Nevertheless, thx a lot for your contribution :heart:

**Comment by Mickael-Martin** [comment]
My patch is already in the "else" block, so I think I don't understand what you explain...

File: src/yunohost/app.py [hunk 1]
```diff
@@ -1901,10 +1901,10 @@ def _fetch_app_from_git(app):
                 # we will be able to use it. Without this option all the history
                 # of the submodules repo is downloaded.
                 subprocess.check_call([
-                    'git', 'clone', '--depth=1', '--recursive', url,
+                    'git', 'clone', '--recursive', url,
```
Comment (alexAubin) [inline src/yunohost/app.py]: Can you elaborate on the removal of `--depth=1` here ? I don't understand how that's related to the issue :/

File: src/yunohost/app.py [hunk 2]
```diff
@@ -1901,10 +1901,10 @@ def _fetch_app_from_git(app):
...
                     extracted_app_folder])
                 subprocess.check_call([
-                    'git', 'reset', '--hard', branch
+                    'git', 'reset', '--hard', 'remotes/origin/'+ branch
```
Comment (alexAubin) [inline src/yunohost/app.py]: Eh I'm surprised that just using the branch name doesn't work ? For this as well, I don't understand how that relates to the Github/Not-Github thing ?

**Comment by Psycojoker** [comment]
> My patch is already in the "else" block, so I think I don't understand what you explain...

Indeed, I've read too fast, sorry about that ^^'

File: src/yunohost/app.py (hunk 1 above)
Comment (Mickael-Martin) [reply src/yunohost/app.py]: or more simple : git clone -b <branch> <remote_repo>

Comment (Mickael-Martin) [reply src/yunohost/app.py]: An error of me, sorry.

File: src/yunohost/app.py (hunk 2 above)
Comment (Mickael-Martin) [reply src/yunohost/app.py]: Actual scenario (not working)
git clone --depth=1 --recursive https://framagit.org/Mickael-Martin/zabbix_ynh
Cloning into 'zabbix_ynh'...
remote: Enumerating objects: 26, done.
remote: Counting objects: 100% (26/26), done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 26 (delta 3), reused 18 (delta 0)
Unpacking objects: 100% (26/26), done.
@mike:~# cd zabbix_ynh/
@mike:~/zabbix_ynh# git branch -a -v
* master                fe30b28 Update post_user_delete
  remotes/origin/HEAD   -> origin/master
  remotes/origin/master fe30b28 Update post_user_delete
@mike:~/zabbix_ynh# git reset --hard testing
fatal: ambiguous argument 'testing': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

Solution 1 (my first proposition) : 
git clone --recursive https://framagit.org/Mickael-Martin/zabbix_ynh
Cloning into 'zabbix_ynh'...
remote: Enumerating objects: 95, done.
remote: Counting objects: 100% (95/95), done.
remote: Compressing objects: 100% (47/47), done.
remote: Total 344 (delta 56), reused 82 (delta 48)
Receiving objects: 100% (344/344), 68.62 KiB | 0 bytes/s, done.
Resolving deltas: 100% (215/215), done.
@mike:~# cd zabbix_ynh/
@mike:~/zabbix_ynh# git branch -a -v
* master                 fe30b28 Update post_user_delete
  remotes/origin/HEAD    -> origin/master
  remotes/origin/master  fe30b28 Update post_user_delete
  remotes/origin/testing fe30b28 Update post_user_delete
@mike:~/zabbix_ynh# git reset --hard remotes/origin/testing
HEAD is now at fe30b28 Update post_user_delete

Solution 2, maybe more efficient
git clone -b testing --single-branch https://framagit.org/Mickael-Martin/zabbix_ynh --depth 1 --recursive
Cloning into 'zabbix_ynh'...
remote: Enumerating objects: 26, done.
remote: Counting objects: 100% (26/26), done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 26 (delta 3), reused 18 (delta 0)
Unpacking objects: 100% (26/26), done.
@mike:~# cd zabbix_ynh/
@mike:~/zabbix_ynh# git branch -a -v
* testing                fe30b28 Update post_user_delete
  remotes/origin/testing fe30b28 Update post_user_delete

**Comment by alexAubin** [comment]
Closing (c.f. https://github.com/YunoHost/yunohost/pull/615 )

**Pull request closed without merging**